### ISSUE TITLE
DDLDatabase failed to parse MySQL NOW(6)

### ISSUE THREAD

**Issue by messense** [jOOQ/jOOQ#9011]
```
$ mvn generate-sources
[INFO] Scanning for projects...
[INFO]
[INFO] ----------------------< org.jooq.trial:jooq-mcve >----------------------
[INFO] Building jOOQ MCVE 1.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- jooq-codegen-maven:3.11.11:generate (java-generator) @ jooq-mcve ---
[INFO] No <inputCatalog/> was provided. Generating ALL available catalogs instead.
[INFO] License parameters
[INFO] ----------------------------------------------------------
[INFO]   Thank you for using jOOQ and jOOQ's code generator
[INFO]
[INFO] Database parameters
[INFO] ----------------------------------------------------------
[INFO] Reading from: src/main/resources/db/migration
[INFO] Reading from: src/main/resources/db/migration/V1__initialise_database.sql [*]
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jooq.tools.reflect.Reflect (file:/Users/lusheng.lls/.m2/repository/org/jooq/jooq/3.11.11/jooq-3.11.11.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class)
WARNING: Please consider reporting this to the maintainers of org.jooq.tools.reflect.Reflect
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[ERROR] An exception occurred while parsing script source : src/main/resources/db/migration/*. Please report this error to https://github.com/jOOQ/jOOQ/issues/new
org.jooq.impl.ParserException: Token ')' expected: [8:38] ...  value INT,
  created_at DATETIME(6) DEFAULT NOW([*]6),

  CONSTRAINT pk_test PRIMARY KEY (id)
);

    at org.jooq.impl.ParserContext.expected (ParserImpl.java:8485)
    at org.jooq.impl.ParserImpl.parse (ParserImpl.java:8052)
    at org.jooq.impl.ParserImpl.parse (ParserImpl.java:8047)
    at org.jooq.impl.ParserImpl.parseTerm (ParserImpl.java:4519)
    at org.jooq.impl.ParserImpl.parseUnaryOps (ParserImpl.java:4208)
    at org.jooq.impl.ParserImpl.parseExp (ParserImpl.java:4191)
    at org.jooq.impl.ParserImpl.parseFactor (ParserImpl.java:4174)
    at org.jooq.impl.ParserImpl.parseSum (ParserImpl.java:4159)
    at org.jooq.impl.ParserImpl.parseCollated (ParserImpl.java:4142)
    at org.jooq.impl.ParserImpl.parseConcat (ParserImpl.java:4132)
    at org.jooq.impl.ParserImpl.parseCreateTable (ParserImpl.java:2339)
    at org.jooq.impl.ParserImpl.parseCreate (ParserImpl.java:1834)
    at org.jooq.impl.ParserImpl.parseQuery (ParserImpl.java:674)
    at org.jooq.impl.ParserImpl.parse (ParserImpl.java:445)
    at org.jooq.impl.ParserImpl.parse (ParserImpl.java:432)
    at org.jooq.meta.extensions.ddl.DDLDatabase.load (DDLDatabase.java:207)
    at org.jooq.meta.extensions.ddl.DDLDatabase.load (DDLDatabase.java:181)
    at org.jooq.meta.extensions.ddl.DDLDatabase.load (DDLDatabase.java:194)
    at org.jooq.meta.extensions.ddl.DDLDatabase.create0 (DDLDatabase.java:157)
    at org.jooq.meta.AbstractDatabase.create (AbstractDatabase.java:231)
    at org.jooq.meta.AbstractDatabase.create (AbstractDatabase.java:220)
    at org.jooq.meta.AbstractDatabase.getDialect (AbstractDatabase.java:202)
    at org.jooq.codegen.AbstractGenerator.logDatabaseParameters (AbstractGenerator.java:155)
    at org.jooq.codegen.JavaGenerator.generate (JavaGenerator.java:299)
    at org.jooq.codegen.GenerationTool.run (GenerationTool.java:752)
    at org.jooq.codegen.GenerationTool.generate (GenerationTool.java:222)
    at org.jooq.codegen.maven.Plugin.execute (Plugin.java:197)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.255 s
[INFO] Finished at: 2019-08-07T10:17:15+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jooq:jooq-codegen-maven:3.11.11:generate (java-generator) on project jooq-mcve: Error running jOOQ code generation tool: Token ')' expected: [8:38] ...  value INT,
[ERROR]   created_at DATETIME(6) DEFAULT NOW([*]6),
[ERROR]
[ERROR]   CONSTRAINT pk_test PRIMARY KEY (id)
[ERROR] );
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

MCVE repository: https://github.com/messense/jOOQ-mcve/tree/ddl-now-6
Reproduction command: `mvn generate-sources`

DDL SQL:

```sql
CREATE TABLE mcve.test (
  id    INT NOT NULL AUTO_INCREMENT,
  value INT,
  created_at DATETIME(6) DEFAULT NOW(6),

  CONSTRAINT pk_test PRIMARY KEY (id)
);
```

**Comment by knutwannheden** [comment]
Thank you for the detailed report and MCVE!

Indeed. The optional `fsp` parameter of `NOW()` was apparently added in MySQL 5.6. The `DSL#now()` API does not directly support this either.

**Comment by knutwannheden** [comment]
This has been fixed for the 3.12 release.

**Comment by lukaseder** [comment]
Adding the change to the parser grammar, too.
